docs/release/release-tasks.md

```diff
@@ -0,0 +1,384 @@
+3. Update references in introduction.md only on the main branch (drop unsupported versions, add the new release version).
+   <br>Prior art: [Add release 1.2 book link](https://github.com/kubernetes-sigs/cluster-api/pull/6697)
+
+#### Polish release notes
+
+1. Checkout the latest commit on the `release-1.4` branch.
+2. Generate release notes with `make release-notes`.
+3. Finalize the release notes:
+    1. Copy & paste the release notes into a hackmd (makes collaboration very easy).
+    2. Pay close attention to the `## :question: Sort these by hand` section, as it contains items that need to be manually sorted.
+    3. Ensure consistent formatting of entries (e.g. prefix (see [v1.2.0](https://github.com/kubernetes-sigs/cluster-api/releases/tag/v1.2.0) release notes)).
+    4. Merge dependency bump PR entries for the same dependency into a single entry.
+    5. Move minor changes into a single line at the end of each section.
+    6. Write highlights section based on the initial release notes doc.
+    7. Add Kubernetes version support section.
+4. Iterate until the GA release (the release note tool can be run starting from a specific commit).
+
+#### Change production branch in Netlify to the new release branch
+
+The goal of this task to make the book for the current release available under `https://cluster-api.sigs.k8s.io`.
+
+Someone with access to Netlify should:
+1. Change production branch in Netlify to `release-1.4` to make the book available under `https://cluster-api.sigs.k8s.io`.
+2. Re-deploy via the Netlify UI.
+
+#### Update clusterctl links in the quickstart 
+
+The goal of this task is to ensure the quickstart has links to the latest `clusterctl` binaries.
+
+1. Update clusterctl links in the quickstart (on main and cherry-pick onto release-1.4).
+   <br>Prior art: [Update clusterctl version to v1.2.x in quick start](https://github.com/kubernetes-sigs/cluster-api/pull/6716)
+
+**Note**: The PR for this should be merged after the minor release has been published.
+
+#### Continuously: Communicate key dates to the community
+
+The goal of this task is to ensure all stakeholders are informed about the current release cycle.
+
+Information can be distributed via: (TBD)
+* `sig-cluster-lifecycle` mailing list
+* Slack
+* Office hours
+* ClusterAPI book
+* ...
+
+Relevant information includes: (TBD)
+* Beta, RC, GA release
+* Start of code freeze
+* Implementation progress
+* ...
+
+Stakeholders are: (TBD)
+* End users of ClusterAPI
+* Contributors to core ClusterAPI
+* Provider implementers
+* ...
+
+## CI Signal/Bug Triage/Automation Manager
+
+### Responsibilities
+
+* Signal:
+    * Responsibility for the quality of the release
+    * Continuously monitor CI signal, so a release can be cut at any time
+    * Add CI signal for new release branches
+* Bug Triage:
+    * Make sure blocking issues and bugs are triaged and dealt with in a timely fashion
+* Automation
+    * Help improve release automation and tools
```

Comment:
I didn't see anything about the automation aspect in the release task issues or below, we should probably include it so it doesn't get lost

Reply:
> * This document focuses on tasks that are done for every release. One-time improvement tasks are out of scope.

My idea was to focus this doc on recurring tasks. Improvements would be separate issues as they are essentially one-time improvements.

Not sure how to frame a task where we improve release tooling in every release cycle (similar for other improvement tasks).

IIRC the k/k sig-release is essentially split in one part which does the continues release stuff and one part which improves release tooling etc.

(cc @cpanato to correct me if I'm wrong)

Reply:
I'll add it to Continuously like the points for communication manager, but I think specific one-time improvements should be tracked in issues and linked in an per-release-cycle umbrella issue